### PR TITLE
Make CodeGeneration depend on swift-syntax 509.0.0

### PR DESCRIPTION
**Pull request by ahoppen**
Having CodeGeneration depend on a stable version of swift-syntax makes rebasing changes a lot easier because CodeGeneration no longer depends on itself.

### CODE REVIEW

File: CodeGeneration/Package.swift [hunk 1]
```diff
@@ -12,17 +12,7 @@ let package = Package(
...
-    // If you are making significant changs to `CodeGeneration` locally and want
-    // to avoid breaking the build of `CodeGeneration` itself by generating new
-    // files in the parent swift-syntax package, it is encouraged to change the
-    // dependency to the following. That way `CodeGeneration` has its own
-    // checkout of swift-syntax that is unaffected by the newly generated files.
-    // Be sure to revert the change before committing your changes.
-    //
-    // .package(url: "https://github.com/apple/swift-syntax", branch: "main")
-    .package(path: "..")
+    .package(url: "https://github.com/apple/swift-syntax", from: "509.0.0")
```
Comment (kimdv) [inline CodeGeneration/Package.swift:15]: How does this work in the CI?

**Review by kimdv — APPROVED** [review]

File: CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/RawSyntaxNodesFile.swift [hunk 2]
```diff
@@ -241,12 +241,12 @@ func rawSyntaxNodesFile(nodesStartingWith: [Character]) -> SourceFileSyntax {
 
           for (index, child) in node.children.enumerated() {
             try VariableDeclSyntax("public var \(child.varOrCaseName.backtickedIfNeeded): Raw\(child.buildableType.buildable)") {
-              let exclamationMark = child.isOptional ? nil : TokenSyntax.exclamationMarkToken()
+              let exclamationMark = child.isOptional ? "" : "!"
 
               if child.syntaxNodeKind == .syntax {
-                ExprSyntax("layoutView.children[\(raw: index)]\(exclamationMark)")
+                ExprSyntax("layoutView.children[\(raw: index)]\(raw: exclamationMark)")
```
Comment (Matejkob) [inline CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/RawSyntaxNodesFile.swift:247]: I'm wondering why this change was made. Weren't we planning to phase out the use of `raw:` interpolation?

**Comment by ahoppen** [comment]
https://github.com/apple/swift/pull/68763

@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by kimdv** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by kimdv — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Linux
